### ISSUE TITLE
Possibility to return alternative success `HRESULT` values

### ISSUE THREAD

**Issue by kaivol** [microsoft/windows-rs#1632]
### Which crate is this about?

windows

### Crate version

0.34.0

### Summary

I'm trying to implement an custom explorer context menu entry, and in the progress I came across the  [`IEnumExplorerCommand::Next`](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ienumexplorercommand-next) method. 
According to the documentation `Next` only returns `S_OK` on success, and consequently has the return type `Result<()>` in [windows-rs](https://microsoft.github.io/windows-docs-rs/doc/windows/Win32/UI/Shell/trait.IEnumExplorerCommand_Impl.html), but I suspect that the documentation is incorrect (see e.g. this [sample](https://github.com/microsoft/Windows-classic-samples/blob/67a8cddc25880ebc64018e833f0bf51589fd4521/Samples/Win7Samples/winui/shell/shellextensibility/explorerdataprovider/FVCommands.cpp#L87)). 

Is there some workaround that allows to return other `HRESULT` values like `S_FALSE`?

**Comment by ishitatsuyuki** [comment]
Duplicate of #1065.

In the current version, you need to add a `PreserveSig` entry to https://github.com/microsoft/win32metadata/blob/master/generation/WinSDK/emitter.settings.rsp in order to disable signature transformation.

**Comment by riverar** [comment]
Yep, we need to fix this up in metadata. Transferring over there.
